plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryClient.java

```diff
@@ -175,6 +189,17 @@ TableInfo getTable(TableId remoteTableId)
         return bigQuery.getTable(remoteTableId);
     }
 
+    TableInfo getCachedTable(ReadSessionCreatorConfig config, String query, TableInfo table)
+    {
+        try {
+            return destinationTableCache.get(query,
```

Comment:
just noting that the cache key seems to be the entire query string (which can be large).
No changes requested at the moment though.